Accessibility tweaks and patching RTL

clrux opened the pull request:
This work appends several tweaks to [already merged work](https://github.com/edx/edx-platform/commit/f5b166f120c950f0290a6c166dceb756b7dc964e) and provides:

* various accessibility enhancements
* improved RTL support

__Setup:__
To setup this branch for review, please do the following:
* [Setup and enable the shopping cart](https://github.com/edx/edx-platform/wiki/Shopping-Cart)
  * You will need a Django admin user (localhost:8000/admin to login)
* In your Vagrant machine (not in the platform)...
  * Modify `devstack.py` to include the following:
```
    "CC_PROCESSOR": {
        "CyberSource2": {
            "ACCESS_KEY": "{ping me on HipChat for this}", 
            "PROFILE_ID": "{ping me on HipChat for this}", 
            "PURCHASE_ENDPOINT": "{ping me on HipChat for this}", 
            "SECRET_KEY": "{ping me on HipChat for this}"
        }
    }, 
    "CC_PROCESSOR_NAME": "CyberSource2",
```

__Browser testing:__

This code has been tested in the following browsers:
```
| Platform | Browser | Version | Passed | 
|----------|---------|---------|--------|
| Mac      | Chrome  | 43      | Yes    |
|          | Safari  | 8       | Yes    |
|          | Firefox | 24      | Yes    |
| Windows  | IE      | 10      | Yes    |
|          |         | 11      | Yes    |
|          | Firefox | 37      | Yes    |
```

---

__Reviewers:__

* @cptvitamin Mark Sadecki (accessibility)
* @frrrances Frances Botsford (FED and Sass)

frrrances commented on lms/static/sass/views/_shoppingcart.scss:
```diff
@@ -225,7 +225,7 @@ $light-border: 1px solid $gray-l5;
   }
 
-  ul.steps {
+  ol.steps {
```
Does this need to have an html selector at all?

clrux replied:
No it doesn't. I changed it to be correct, but should have just removed it entirely.

frrrances commented on lms/templates/shoppingcart/shopping_cart.html:
```diff
@@ -73,10 +73,14 @@
-          <h1>${ course.display_name }</h1><span class="pull-right">${course.start_datetime_text()} -  ${course.end_datetime_text()}</span>
+          <p class="course-meta-info" aria-describedby="course-title">
+            <span class="course-dates-title">${_('Course Dates:')}</span>
+            <span class="course-display-dates clearfix">${ course.start_datetime_text() } - ${ course.end_datetime_text() }</span>
```
Can you move this clearfix into the sass?

frrrances commented on lms/static/sass/views/_shoppingcart.scss:
```diff
@@ -303,6 +309,34 @@ $light-border: 1px solid $gray-l5;
+        color: $dark-gray2;
+      }
+
+      .course-title-info {
```
Can you move these layout bits above the styles for the inner bits (insert at line 311)?

frrrances commented:
@clrux Code looks good - just a few small nits - but I can't get my local to run it and check the visuals... can we take a few minutes together tomorrow morning and get this to done?

clrux commented:
@frrrances I've addressed your 'nits'. Feel free to drop by today and I can run you through the UI.

frrrances commented on lms/static/sass/views/_shoppingcart.scss:
```diff
@@ -303,6 +309,38 @@ $light-border: 1px solid $gray-l5;
+        @include clearfix();
+      }
+
+      .course-title-info {
```
Can you move these layout pieces (.course-title-info and .course-meta-info) to line 311?

frrrances commented on lms/static/sass/views/_shoppingcart.scss:
```diff
@@ -1036,7 +1074,7 @@ $light-border: 1px solid $gray-l5;
   }
 
-  ul.steps, a.blue.pull-right, .bordered-bar span.pull-right, .left.nav-global.authenticated {
+  ol.steps, a.blue.pull-right, .bordered-bar span.pull-right, .left.nav-global.authenticated {
```
remove element selector

frrrances commented:
@clrux Couple more comments and I'm ready to look over your shoulder... also I noticed the same layout on the receipt page - it would be nice to match the new markup there as well: 

![screen shot 2015-06-16 at 9 17 19 am](https://cloud.githubusercontent.com/assets/4327102/8183832/a07e0330-1408-11e5-9910-efbff966a1d1.png)

clrux commented:
@frrrances I've addressed your feedback. I've also attached a screenshot of the relevant portion of the receipt page. It takes up two pieces of paper so I've created a ticket for us to streamline the receipt and save the earth.

![screen shot 2015-06-16 at 10 58 27 am](https://cloud.githubusercontent.com/assets/2112024/8186161/a5a68270-1416-11e5-9ecb-d7089abbfd6c.png)

frrrances commented:
Thanks @clrux! :+1: once tests are passing.
